src/game_api/layer.hpp

```diff
@@ -1,28 +1,96 @@
 #pragma once
 #include "entity.hpp"
 
+struct Entities
+{
+    Entity** ent_list;
+    uint32_t* uids;
+    uint32_t cap;
+    uint32_t size;
+
+    constexpr Entity** begin()

```

Comment:
For even easier usage we could return a `struct EntitiesIterable;` that has the `begin` and `end` functions defined as this. And then do the same with `UidsIterable` doing the same so you can iterate either entities or uids in a range-based for-loop.